runelite-client/src/main/java/net/runelite/client/eventbus/EventBus.java

```diff
@@ -43,6 +44,14 @@ private CompositeDisposable getCompositeDisposable(@NonNull Object object)
 	// Subscribe on lifecycle (for example from plugin startUp -> shutdown)
 	public <T> void subscribe(Class<T> eventClass, @NonNull Object lifecycle, @NonNull Consumer<T> action)
 	{
+		for (Map.Entry<Object, Object> subscriptions : subscriptionList.entrySet())
+		{
+			if (subscriptions.getKey() == lifecycle && subscriptions.getValue() == eventClass)
```

Comment:
this should use `.equals`

or `if (subscriptionList.containsKey(lifecycle) && eventclass.equals(subscriptionList.get(lifecycle)))